src/DotNetty.Transport/Channels/DefaultChannelId.cs

```diff
@@ -0,0 +1,419 @@
+            {
+                return 1;
+            }
+
+            // Must be EUI-48 or longer.
+            if (candidate.Length < 6)
+            {
+                return 1;
+            }
+
+            // Must not be filled with only 0 and 1.
+            bool onlyZeroAndOne = true;
+            foreach (byte b in candidate)
+            {
+                if (b != 0 && b != 1)
+                {
+                    onlyZeroAndOne = false;
+                    break;
+                }
+            }
+
+            if (onlyZeroAndOne)
+            {
+                return 1;
+            }
+
+            // Must not be a multicast address
+            if ((candidate[0] & 1) != 0)
+            {
+                return 1;
+            }
+
+            // Prefer globally unique address.
+            if ((current[0] & 2) == 0)
+            {
+                if ((candidate[0] & 2) == 0)
+                {
+                    // Both current and candidate are globally unique addresses.
+                    return 0;
+                }
+                else
+                {
+                    // Only current is globally unique.
+                    return 1;
+                }
+            }
+            else
+            {
+                if ((candidate[0] & 2) == 0)
+                {
+                    // Only candidate is globally unique.
+                    return -1;
+                }
+                else
+                {
+                    // Both current and candidate are non-unique.
+                    return 0;
+                }
+            }
+        }
+
+        static int CompareAddresses(IPAddress current, IPAddress candidate)
+        {
+            return ScoreAddress(current) - ScoreAddress(candidate);
+        }
+
+        static int ScoreAddress(IPAddress addr)
+        {
+            if (IPAddress.IsLoopback(addr))
+            {
+                return 0;
+            }
+            if (addr.IsIPv6Multicast)
+            {
+                return 1;
+            }
+            if (addr.IsIPv6LinkLocal)
+            {
+                return 2;
+            }
+            if (addr.IsIPv6SiteLocal)
+            {
+                return 3;
+            }
+
+            return 4;
+        }
+
+        public string AsShortText()
+        {
+            string asShortText = this.shortValue;
+            if (asShortText == null)
+            {
+                this.shortValue = asShortText = ByteBufferUtil.HexDump(this.data, MachineIdLen + ProcessIdLen + SequenceLen + TimestampLen, RandomLen);
+            }
+
+            return asShortText;
+        }
+
+        public string AsLongText()
+        {
+            string asLongText = this.longValue;
+            if (asLongText == null)
+            {
+                this.longValue = asLongText = this.NewLongValue();
+            }
+            return asLongText;
+        }
+
+        public int CompareTo(IChannelId other)
+        {
+            return 0;
```

Comment:
it's the same like in the original code, https://github.com/netty/netty/blob/master/transport/src/main/java/io/netty/channel/DefaultChannelId.java#L448

for reference, otherwise it's not hard to implement it.